src/common-utils/scripts/bash_theme_snippet.sh

```diff
@@ -23,3 +22,20 @@ __bash_prompt() {
 }
 __bash_prompt
 export PROMPT_DIRTRIM=4
+
+# Function to set the terminal title to the command being executed
+preexec() {
+    local cmd=$(history 1 | sed 's/^[ ]*[0-9]*[ ]*//')
+    echo -ne "\033]0;${USER}@${HOSTNAME}: ${cmd}\007"
+}
+
+# Function to reset the terminal title to the shell type after the command is executed
+precmd() {
+    echo -ne "\033]0;${USER}@${HOSTNAME}: ${SHELL}\007"
+}
+
+# Trap DEBUG signal to call preexec before each command
+trap 'preexec' DEBUG
```

Comment:
Using the `DEBUG` trap can introduce performance overhead because it triggers the `preexec` function before every command.

Can we wrap these new changes within the condition `"${TERM_PROGRAM}" = "codespaces"` ? If we could have a new TERM_PROGRAM for `workspace` then it's even better.

Reply:
Added a check for `xterm` which is what we use for workspace